### PR TITLE
k8s: add feature toggle and stub to save dashboards k8s 

### PR DESCRIPTION
**Pull request by ryantxu**
**What is this feature?**

This adds a feature toggle and a stub DashboardService implementation that will write to k8s

**Why do we need this feature?**

This gives us a path to start exploring k8s flavor while keeping existing service in place

**Who is this feature for?**

¯\\_(ツ)_/¯

### CODE REVIEW

File: pkg/services/store/k8saccess/dashboard_service.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/store/entity"
+)
+
+type k8sDashboardService struct {
+	orig  dashboards.DashboardService
+	store entity.EntityStoreServer
```
Comment (toddtreece) [inline pkg/services/store/k8saccess/dashboard_service.go:14]:
```go
    if features.IsEnabled(featuremgmt.FlagK8sDashboards) {
		if k8s.GetSystemClient() == nil {
			panic("k8s dashboards requires the k8s client registered")
		}
		return k8saccess.NewDashboardService(svc, store)
	}
	return svc
```
i'm not following how either the original dashboard store or the entity store will be used here. based on this feature flag i was assuming this was writing to k8s, and then the k8s controller would write to the original store?

**Comment by sdboyer** [comment]
i would really prefer you attach this to the kind system, as we discussed yesterday

File: pkg/services/store/k8saccess/dashboard_service.go (hunk 1 above)
Comment (ryantxu) [reply pkg/services/store/k8saccess/dashboard_service.go:14]: yes -- we just need to intercept the write/modify/delete paths, the rest still has to be called from the original service

**Comment by ryantxu** [comment]
@sdboyer this is just setting up a feature flag and an alternative service, not even sure how the kindsys would help here 🤷

**Comment by ryantxu** [comment]
This PR does not intend to actually save anything... just get the scaffolding in place to start working

**Review by mildwonkey — APPROVED** [review]
👍 LGTM, things still work as expected in "regular" mode, let's do some experimenting!

**Comment by sdboyer** [comment]
apologies - after some discussion with @ryantxu i now see i misunderstood the focus of this PR (sorry!)

there is a path to kindsys helping here - quite a bit - but those are details i hashed out with with @toddtreece and @jalevin, too much to pack in here

**Comment by ryantxu** [comment]
> there is a path to kindsys helping here

👍 we will explore this in the POC, for now we will punt on generic middleware and just focus on the save loop

**Pull request merged**